spec.go

```diff
@@ -311,26 +311,47 @@ func createLibcontainerMount(cwd string, m specs.Mount) *configs.Mount {
 	}
 }
 
-func createCgroupConfig(name string, spec *specs.Spec) (*configs.Cgroup, error) {
+func createCgroupConfig(name string, useSystemdCgroup bool, spec *specs.Spec) (*configs.Cgroup, error) {
 	var (
 		err          error
 		myCgroupPath string
 	)
 
+	c := &configs.Cgroup{
+		Resources: &configs.Resources{},
+	}
+
 	if spec.Linux.CgroupsPath != nil {
 		myCgroupPath = libcontainerUtils.CleanPath(*spec.Linux.CgroupsPath)
+	}
+
+	if useSystemdCgroup {
+		if myCgroupPath == "" {
+			c.Parent = "system.slice"
+			c.ScopePrefix = "runc"
+			c.Name = name
+		} else {
+			// Parse the path from expected "slice:prefix:name"
+			// for e.g. "system.slice:docker:1234"
+			parts := strings.Split(myCgroupPath, ":")
```

Comment:
This is not how docker works today. Refer daemon/oci_linux.go:createSpec. Couple of issues:
1. CgroupParent and containerID are combined using a filepath.Join. Not by a colon separator.
2. There's no ScopePrefix appended to the CgroupsPath.

Are the requirements about cgroups path being in "slice:prefix:name" format part of the spec? If yes, we will need docker changes to go along with this commit.